### PR TITLE
Changes examples to use new node builder interface.

### LINKED ISSUE #28: API service example code out of date again
`core.NewNodeBuilder` no longer exists, and the `"code.google.com/p/go.net/context"` import URL should be updated.

### PR DESCRIPTION
**Pull request by eelcocramer**
Fixes #28

### CODE REVIEW

**Comment by jbenet** [comment]
@eelcocramer thanks

please use gofmt -- your PR has a bunch of changes that are just changing whitespace from tabs to spaces, something that gofmt is against.

**Comment by eelcocramer** [comment]
I already thought that might be a problem :-p I'm new to go and didn't know about gofmt. I've fixed the files and added the commit to the PR.

Thanks.

File: examples/api/demoapp/demo.go [hunk 1]
```diff
@@ -26,13 +26,16 @@ func CountChars(r io.Reader) map[byte]int {
...
-	r := fsrepo.At("~/.ipfs")
-	if err := r.Open(); err != nil {
+	r, err := fsrepo.Open("~/.ipfs")
+	if err != nil {
 		return nil, err
 	}
 
-	builder := core.NewNodeBuilder().Online().SetRepo(r)
-	return builder.Build(context.Background())
+	cfg := new(core.BuildCfg)
```
Comment (whyrusleeping) [inline examples/api/demoapp/demo.go]: I would prefer to use the composite construction here:

``` go
cfg := &core.BuildCfg{
    Repo: r,
    Online: true,
}
```

**Comment by eelcocramer** [comment]
That is more elegant.

File: examples/api/demoapp/demo.go [hunk 2]
```diff
@@ -1,15 +1,15 @@
 package main
 
 import (
-	"bufio"
+    "bufio"
```
Comment (jbenet) [inline examples/api/demoapp/demo.go]: `gofmt`. please run `gofmt` every time. add it to your editor.

File: examples/api/demoapp/readme.md [hunk 3]
```diff
@@ -10,15 +10,15 @@ To Start off, lets get some imports:
...
-	"code.google.com/p/go.net/context"
+    "bufio"
+    "fmt"
+    "io"
+    "os"
+
+    "github.com/ipfs/go-ipfs/core"
+    "github.com/ipfs/go-ipfs/core/coreunix"
+    "github.com/ipfs/go-ipfs/repo/fsrepo"
+    "code.google.com/p/go.net/context"
```
Comment (jbenet) [inline examples/api/demoapp/readme.md]: go fmt.

Comment (eelcocramer) [inline examples/api/demoapp/readme.md]: So, gofmt does nothing on the readme.

File: examples/api/demoapp/demo.go (hunk 2 above)
Comment (eelcocramer) [inline examples/api/demoapp/demo.go]: Thought I had that figured out. Apparently not. Will add a commit momentarily. Thanks.

**Comment by eelcocramer** [comment]
Ok, format is fixed again. Seems that I imported a vim bundle from the wrong location. I'm ready to "go" now... :-)

File: examples/api/readme.md [hunk 4]
```diff
@@ -63,21 +41,31 @@ import "github.com/ipfs/go-ipfs/repo/fsrepo"
...
 to get this information through, just make an object that implements the
-IpfsRouting interface and pass the builder a RoutingOption for it.
+IpfsRouting interface and pass the build configuration a RoutingOption for it.
+
+### Ipfs Configuration Options
+
+### Context
+If youve never dealt with contexts before, I highly recommend you first go read
+[this wonderful explanation](https://blog.golang.org/context). Now, the context
+we pass into `Build` is the "master" context to the entire ipfs node, cancelling
```
Comment (jbenet) [inline examples/api/readme.md]: `Build` is no longer a thing, right?

**Comment by jbenet** [comment]
> I'm ready to "go" now... :-)

hahaha :) 

LGTM above, except the one stale ref to `Build`

File: examples/api/readme.md (hunk 4 above)
Comment (eelcocramer) [inline examples/api/readme.md]: Missed that one... Thanks.

**Comment by jbenet** [comment]
thanks @eelcocramer

**Comment by eelcocramer** [comment]
Thank you @jbenet for creating ipsf in the first place!

**Pull request merged**